src/util-decode-mime.c

```diff
@@ -950,8 +950,8 @@ static int IsIpv6Host(const uint8_t *urlhost, uint32_t len)
     char tempIp[MAX_IP6_CHARS + 1];
 
     /* Cut off at '/'  */
-    uint32_t i = 0;
-    for (i = 0; i < len && urlhost[i] != 0; i++) {
+    uint8_t i = 0;
+    for (i = 0; i < len && urlhost[i] != 0 && i < UINT8_MAX; i++) {
```

Comment:
I feel this is an anti-pattern. We have a length field, but we ignore it?

Reply:
> I feel this is an anti-pattern. We have a length field, but we ignore it?

How so ?
`i < len` does not look to me like ignoring the length field...

Reply:
whoops misread the diff